v2/component/runtime_info.go

```diff
@@ -71,6 +72,27 @@ type runtimeInfo struct {
 	OutputArtifacts  map[string]*outputArtifact
 }
 
+var paramDelimiterRE = regexp.MustCompile(`"BEGIN-KFP-PARAM[[].+?[]]END-KFP-PARAM"`)
```

Comment:
the difference between `.+?` and `.*` probably needs a comment.
I didn't understand that initially: https://stackoverflow.com/a/14213895/8745218

Reply:
`.+?` or `.*?` is greedy, they match as much as possible, while `.+` or `.*` is non-greedy.
I think it's fairly common knowledge about regular expression. But I guess additional comments won't hurt. :)

Reply:
Thanks. Added comment.